### DISCUSSION TITLE
How to pipe an image to imdisplay (on Windows)?

### DISCUSSION THREAD

**Discussion by totalL2020** [ImageMagick/ImageMagick#2164] [Help] [answered]
I came across the following two examples on ImageMagick's website under _Command Line Processing_:
```
magick logo: gif:- | display gif:-
magick logo: gif:- | display -define stream:buffer-size=0 gif:-
```
But since I'm using Windows I figured that _display_ should be replaced with _imdisplay_ (because that's the only exe I have in the directory besides _magick_). But `magick logo: gif:- | imdisplay gif:-` results in `\Program Files\ImageMagick-7.0.10-Q16-HDRI\gif:- was not found` message box. Removing second `gif:` opens the _imdisplay_ window but with no image inside, which is a progress.

Reading on the tutorial I then thought the issue may be due to the buffer which may delay the image data to be read by _imdisplay_. So I did `magick logo: gif:- | imdisplay -define stream:buffer-size=0 -` and got `\Program Files\ImageMagick-7.0.10-Q16-HDRI\stream:buffer-size=0 was not found` message box. What's more, my command prompt window froze like something's being processed, even Ctrl+C (cancel command) didn't help, which surprised me (it usually does the trick of stopping the "freeze").

Starting a new command prompt, I tried `magick logo: gif:- -define stream:buffer-size=0 | imdisplay -` and got same result (_imdisplay_ opened with no image inside) with exception that I could cancel the "hanging" with Ctrl+C this time.

Thinking that something's wrong with _imdisplay_, I tried piping the image to _mspaint_ instead with `magick logo: gif:- | %windir%\system32\mspaint.exe -` but got `\Program Files\ImageMagick-7.0.10-Q16-HDRI\-.png was not found`  message box.

What is the problem here? How do you pipe an image to _imdisplay_ on Windows?

**Comment by snibgo** [comment, marked as answer]
I don't think imdisplay can read from stdin. Instead, just write to "win:", eg:
```
magick logo: win:
```

**Reply by totalL2020** [reply]
Thank you!

**Reply by PeteMWilson** [reply]
This apparently no longer works in ImageMagick 7.1.1? Was it removed or is it broken?

**Reply by dlemstra** [reply]
See my response in your issue here: https://github.com/ImageMagick/ImageMagick/issues/7609
